webapp/channels/src/components/channel_settings_modal/channel_settings_modal.test.tsx

```diff
@@ -438,4 +439,69 @@ describe('ChannelSettingsModal', () => {
             expect(screen.getByRole('tab', {name: 'access_rules'})).toBeInTheDocument();
         });
     });
+
+    describe('preventClose functionality', () => {
+        it('should pass preventClose prop to GenericModal based on unsaved changes state', async () => {
+            // Mock the InfoTab to simulate unsaved changes
+            const MockInfoTabWithChanges = jest.fn().mockImplementation(({setAreThereUnsavedChanges}) => {
+                // Simulate unsaved changes by calling the setter immediately
+                React.useEffect(() => {
+                    setAreThereUnsavedChanges?.(true);
+                }, [setAreThereUnsavedChanges]);
+
+                return <div data-testid='info-tab-with-changes'>{'Info Tab with unsaved changes'}</div>;
+            });
+
+            jest.doMock('./channel_settings_info_tab', () => MockInfoTabWithChanges);
+
+            renderWithContext(<ChannelSettingsModal {...baseProps}/>);
+
+            // Wait for the modal to load
+            await waitFor(() => {
+                expect(screen.getByRole('dialog')).toBeInTheDocument();
+            });
+
+            const modal = screen.getByRole('dialog');
+
+            // Try to close the modal - it should prevent closing due to unsaved changes
+            const closeButton = screen.getByLabelText(/close/i);
+            act(() => {
+                fireEvent.click(closeButton);
+            });
+
+            // Modal should still be visible (not closed due to unsaved changes)
+            expect(modal).toBeInTheDocument();
+            expect(modal).toBeVisible();
+
+            // Restore the original mock
+            jest.doMock('./channel_settings_info_tab', () => {
+                return function MockInfoTab(): JSX.Element {
+                    return <div data-testid='info-tab'>{'Info Tab Content'}</div>;
+                };
+            });
```

Comment:
Nitpick:

You could add a "setAreThereUnsavedChanges" button to the original mock. That way, you simplify a couple of things:
- No need for a "out of place" effect, which is a bit harder to understand
- A bit more intention revealing since you "click on setAreThereUnsavedChanges"
- No need to restore the mock